master/internal/scheduler/default_resource_provider.go

```diff
@@ -352,6 +352,8 @@ func (d *DefaultRP) receiveAddTask(ctx *actor.Context, msg AddTask) {
 		name = "Unnamed Task"
 	}
 
+	msg.FittingRequirements.DedicatedAgent = d.taskContainerDefaults.NetworkMode.IsHost()
```

Comment:
can also throw this in before the trial messages the rp at all, just seemed a little cleaner to just add it here, rather than needing to add this context to the trial.

also, i'm a little unclear on the k8s story here.

Reply:
blocking: if we do this check here we will end up not allowing multi-slot notebooks (unless they take up the whole agent) in host mode. Could we set this in `trial.go` instead?

k8s completely ignores these fitting requirements and we always set the networking to w/e the default is (really shouldn't be host mode on k8s).